tests/gccgo/task.yaml

```diff
@@ -0,0 +1,13 @@
+summary: Check that snapd builds with gccgo
+prepare: |
+    echo Installing gccgo-6 and pretending it is the default go
+    sudo apt install -y gccgo-6
+    sudo ln -s /usr/bin/go-6 /usr/local/bin/go
+restore: |
+    rm -f /usr/local/bin/go
+    sudo apt-get autoremove -y gccgo-6
+execute: |
```

Comment:
Just to make sure we're testing the right thing, can we please have something like this here
```
go version | grep gccgo
```

I haven't tested. Not sure if gccgo is actually listed on that string, but I'd expect so.